suitcase/utils/__init__.py

```diff
@@ -1,25 +1,219 @@
+from pathlib import Path
+import collections
+import os
+import io
+import event_model
+from bluesky.plans import count
 from ._version import get_versions
 __version__ = get_versions()['version']
 del get_versions
 
 
-# Suitcase subpackages must follow strict naming and interface conventions. The
-# public API should include some subset of the following. Any functions not
-# implemented should be omitted, rather than included and made to raise
-# NotImplementError, so that a client importing this library can immediately
-# know which portions of the suitcase API it supports without calling any
-# functions.
-#
-# def export(...)
-#     ...
-#
-#
-# def ingest(...):
-#     ...
-#
-#
-# def reflect(...):
-#     ...
-#
-#
-# handlers = []
+class SuitcaseUtilsError(Exception):
+    ...
+
+
+class SuitcaseUtilsValueError(SuitcaseUtilsError):
+    ...
+
+
+class SuitcaseUtilsTypeError(SuitcaseUtilsError):
+    ...
+
+
+class SuitcaseUtilsModeError(SuitcaseUtilsError):
```

Comment:
The `SuitcaseUtils` prefix is only needed when we need to distinguish a library-specific subclass of a built-in error type (`ValueError`, `TypeError` above). But we don't need it for fully custom errors -- this can just be `ModeError`, same as event_model has an `UnfilledError`.